core/src/main/java/org/springframework/security/authorization/method/AuthorizationManagerAfterReactiveMethodInterceptor.java

```diff
@@ -151,7 +152,23 @@ private Mono<Object> postAuthorize(Mono<Authentication> authentication, MethodIn
 		MethodInvocationResult invocationResult = new MethodInvocationResult(mi, result);
 		return this.authorizationManager.check(authentication, invocationResult)
 			.switchIfEmpty(Mono.just(new AuthorizationDecision(false)))
-			.flatMap((decision) -> postProcess(decision, invocationResult));
+			.flatMap((decision) -> postProcess(decision, invocationResult))
+			.onErrorResume(AuthorizationDeniedException.class, (denied) -> postProcess(denied, invocationResult));
```

Comment:
Does this also handle if `postProcess` throws a `AuthorizationDeniedException`?

Reply:
Good point. I've updated the logic.